spring-rabbit/src/main/java/org/springframework/amqp/rabbit/listener/DirectMessageListenerContainer.java

```diff
@@ -0,0 +1,226 @@
+	}
+
+	@Override
+	protected void doInitialize() throws Exception {
+		Assert.notNull(getQueueNames(), "queue(s) are required");
+	}
+
+	@Override
+	protected void doStart() throws Exception {
+		super.doStart();
+		if (this.taskExecutor == null) {
+			this.taskExecutor = new SimpleAsyncTaskExecutor((getBeanName() == null ? "container" : getBeanName()) + "-");
+		}
+		AtomicBoolean initialized = new AtomicBoolean();
+		this.taskExecutor.execute(() -> {
+
+				while (!initialized.get() && isRunning()) {
+					try {
+						for (int i = 0; i < DirectMessageListenerContainer.this.consumersPerQueue; i++) {
+							for (String queue : getQueueNames()) {
+								Connection connection = getConnectionFactory().createConnection();
+								Channel channel = connection.createChannel(isChannelTransacted());
+								channel.basicQos(DirectMessageListenerContainer.this.prefetch);
+								RabbitUtils.setPhysicalCloseRequired(true);
+								SimpleConsumer consumer = new SimpleConsumer(channel, queue);
+								DirectMessageListenerContainer.this.consumers.add(consumer);
+								channel.basicConsume(queue, consumer);
+							}
+						}
+					}
+					catch (Exception e) {
+						logger.error("Error creating consumer; retrying in "
+								+ DirectMessageListenerContainer.this.recoveryInterval, e);
+						doShutdown();
+						try {
+							Thread.sleep(DirectMessageListenerContainer.this.recoveryInterval);
+						}
+						catch (InterruptedException e1) {
+							Thread.currentThread().interrupt();
+						}
+						continue; // initialization failed; try again having rested for recovery-interval
+					}
+					initialized.set(true);
+				}
+
+		});
+		if (logger.isInfoEnabled()) {
+			logger.info("Container initialized for " + Arrays.asList(getQueueNames()));
+		}
+	}
+
+	@Override
+	protected void doShutdown() {
+		Assert.state(this.taskExecutor != null, "Cannot shut down if not initialized");
+		for (DefaultConsumer consumer : this.consumers) {
+			try {
+				consumer.getChannel().basicCancel(consumer.getConsumerTag());
+				RabbitUtils.closeChannel(consumer.getChannel());
+			}
+			catch (IOException e) {
+				logger.error("Cancel Error", e);
+			}
+		}
+		this.consumers.clear();
+	}
+
+	private final class SimpleConsumer extends DefaultConsumer {
+
+		private final Log logger = DirectMessageListenerContainer.this.logger;
+
+		private final String queue;
+
+		private final boolean ackRequired;
+
+		private SimpleConsumer(Channel channel, String queue) {
+			super(channel);
+			this.queue = queue;
+			this.ackRequired = !getAcknowledgeMode().isAutoAck() && !getAcknowledgeMode().isManual();
+		}
+
+		@Override
+		public void handleDelivery(String consumerTag, Envelope envelope,
+				BasicProperties properties, byte[] body) throws IOException {
+			MessageProperties messageProperties = DirectMessageListenerContainer.this.messagePropertiesConverter
+					.toMessageProperties(properties, envelope, "UTF-8");
+			messageProperties.setMessageCount(0);
```

Comment:
I think I'd rather change it to an int and let it default to 0 - probably a "less-likely" breaking change - compile time Vs. runtime.